onnxruntime/core/session/custom_ops.cc

```diff
@@ -381,6 +381,7 @@ ORT_API_STATUS_IMPL(OrtApis::KernelContext_GetResource, _In_ const OrtKernelCont
   API_IMPL_END
 };
 
+#if !defined(ORT_MINIMAL_BUILD) || defined(ORT_MINIMAL_BUILD_CUSTOM_OPS)
```

Comment:
Would it be slightly more convenient to define something like this:

`#define CUSTOM_OP_API_ENABLED !defined(ORT_MINIMAL_BUILD) || defined(ORT_MINIMAL_BUILD_CUSTOM_OPS)`

and with that we can exclude the implementation for any/all of the OrtApis::X_Y methods that implement the custom op API, starting with the new one?

I would suggest doing the exclusion inside one declaration of the function so there's no chance the args will get out of sync.

e.g. using one of the existing custom op API implementations

```
#define CUSTOM_OP_API_ENABLED !defined(ORT_MINIMAL_BUILD) || defined(ORT_MINIMAL_BUILD_CUSTOM_OPS)

#if CUSTOM_OP_API_ENABLED 
// static function to minimize binary size by not duplicating the CreateStatus code in multiple places
static OrtStatusPtr CustomOpsDisabledError() {
  return OrtApis::CreateStatus(ORT_NOT_IMPLEMENTED, "Custom Ops API is not enabled in this build");
}
#endif

ORT_API_STATUS_IMPL(OrtApis::ShapeInferContext_GetInputCount, _In_ const OrtShapeInferContext* context, _Out_ size_t* out) {
#if CUSTOM_OP_API_ENABLED
  API_IMPL_BEGIN
  *out = context->GetInputCount();
  return nullptr;
  API_IMPL_END
#else
  return CustomOpsDisabledError();
#endif
}
```

Reply:
It is decent, except that we gonna have a "context unreferenced ..." error while building minimal?